packages/cli/src/cli/provider/action.provider.tag.ts

```diff
@@ -0,0 +1,93 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import { Aws, LogConfig, TileMetadataTag } from '@basemaps/lambda-shared';
+import {
+    CommandLineAction,
+    CommandLineFlagParameter,
+    CommandLineIntegerParameter,
+    CommandLineStringParameter,
+} from '@rushstack/ts-command-line';
+import { invalidateCache } from '../util';
+import { validateProvider, printProvider } from './provider.util';
+import * as c from 'ansi-colors';
+
+export class ProviderUpdateTagAction extends CommandLineAction {
+    private version: CommandLineIntegerParameter;
+    private tag: CommandLineStringParameter;
+    private commit: CommandLineFlagParameter;
+
+    public constructor() {
+        super({
+            actionName: 'tag',
+            summary: 'Tag a version',
+            documentation: '',
+        });
+    }
+
+    protected onDefineParameters(): void {
+        this.version = this.defineIntegerParameter({
+            argumentName: 'VERSION',
+            parameterLongName: '--version',
+            parameterShortName: '-v',
+            description: 'Version ID',
+            required: false,
+        });
+
+        const validTags = Object.values(TileMetadataTag).filter((f) => f != TileMetadataTag.Head);
+        this.tag = this.defineStringParameter({
+            argumentName: 'TAG',
+            parameterLongName: '--tag',
+            parameterShortName: '-t',
+            description: `tag name  (options: ${validTags.join(', ')})`,
+            required: false,
+        });
+
+        this.commit = this.defineFlagParameter({
+            parameterLongName: '--commit',
+            description: 'Commit to database',
+            required: false,
+        });
+    }
+
+    protected async onExecute(): Promise<void> {
+        const tagInput = this.tag.value!;
+        const version = this.version.value!;
+
+        let tag: TileMetadataTag;
+        switch (tagInput) {
```

Comment:
We now do this in multiple places, possibly should export a function to do this, so we dont forget to update it in both places.